inst/templates/ad_adlb.R

```diff
@@ -0,0 +1,328 @@
+  "CREAT",   "CREAT",  "Creatinine (umol/L)", 14,
+  "EOS",     "EOS",    "Eosinophils (GI/L)", 15,
+  "GGT",     "GGT",    "Gamma Glutamyl Transferase (U/L)", 16,
+  "GLUC",    "GLUC",   "Glucose (mmol/L)", 17,
+  "HBA1C",   "HBA1C",  "Hemoglobin A1C (1)", 18,
+  "HCT",     "HCT",    "Hematocrit (1)", 19,
+  "HGB",     "HGB",    "Hemoglobin (mmol/L)", 20,
+  "K",       "POTAS",  "Potassium (mmol/L)", 21,
+  "KETONES", "KETON",  "Ketones", 22,
+  "LYM",     "LYMPH",  "Lymphocytes (GI/L)", 23,
+  "MACROCY", "MACROC", "Macrocytes", 24,
+  "MCH",     "MCH",    "Ery. Mean Corpuscular Hemoglobin (fmol(Fe))", 25,
+  "MCHC",    "MCHC",   "Ery. Mean Corpuscular HGB Concentration (mmol/L)", 26,
+  "MCV",     "MCV",    "Ery. Mean Corpuscular Volume (f/L)", 27,
+  "MICROCY", "MICROC", "Microcytes", 28,
+  "MONO",    "MONO",   "Monocytes (GI/L)", 29,
+  "PH",      "PH",     "pH", 30,
+  "PHOS",    "PHOS",   "Phosphate (mmol/L)", 31,
+  "PLAT",    "PLAT",   "Platelet (GI/L)", 32,
+  "POIKILO", "POIKIL", "Poikilocytes", 33,
+  "POLYCHR", "POLYCH", "Polychromasia", 34,
+  "PROT",    "PROT",   "Protein (g/L)", 35,
+  "RBC",     "RBC",    "Erythrocytes (TI/L)", 36,
+  "SODIUM",  "SODIUM", "Sodium (mmol/L)", 37,
+  "SPGRAV",  "SPGRAV", "Specific Gravity", 38,
+  "TSH",     "TSH",    "Thyrotropin (mU/L)", 39,
+  "URATE",   "URATE",  "Urate (umol/L)", 40,
+  "UROBIL",  "UROBIL", "Urobilinogen", 41,
+  "VITB12",  "VITB12", "Vitamin B12 (pmol/L)", 42,
+  "WBC",     "WBC",    "Leukocytes (GI/L)", 43
+)
+
+
+# ---- Derivations ----
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- vars(TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+adlb <- lb %>%
+
+  # Join ADSL with LB (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+
+  # Calculate ADT, ADY
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = LBDTC,
+    flag_imputation = FALSE
+  ) %>%
+
+  derive_var_ady(reference_date = TRTSDT, date = ADT)
+
+# Get WBC
+# Uncomment when derive_param_wbc_abs() is merged (PR #1035)
+# adlb <- adlb %>%
+#   derive_param_wbc_abs(#dataset = input,
+#                        by_vars = vars(USUBJID, VISIT),
+#                        set_values_to = vars(PARAMCD = "LYMPHSI",
+#                                           DTYPE = "CALCULATION"),
+#                        filter_diff = LBSTRESU == "fraction of 1",
+#                        wbc_code = "WBC",
+#                        diff_code = "LYMLE")
+
+adlb <- adlb %>%
+
+  # Add PARAMCD PARAM and PARAMN - from LOOK-UP table
+  # Replace with PARAMCD lookup function
+  derive_vars_merged(
+    dataset_add = param_lookup,
+    new_vars = vars(PARAMCD, PARAM, PARAMN),
+    by_vars = vars(LBTESTCD)
+  ) %>%
+
+  # Calculate PARCAT1 AVAL AVALC ANRLO ANRHI
+  mutate(
+    PARCAT1 = LBCAT,
+    AVAL  = LBSTRESN,
+    AVALC = LBSTRESC,
+    ANRLO = LBSTNRLO,
+    ANRHI = LBSTNRHI
+  )
+
+# get visit info
```

Comment:
I was chatting with another colleague and he mentioned sometimes we can have summary records derived for ADLB (e.g. average value of a parameter for each visit). Maybe we can add it in the workflow too if this is common for studies. For example, using `derive_summary_records()` like the example given in the ADVS workflow.